vignettes/adrs.Rmd

```diff
@@ -184,19 +189,25 @@ dataset_vignette(
 )
 ```
 
-### Derive `ANL01FL`
+### Derive `ANL01FL` {#anl01fl}
 
-When deriving `ANL01FL` this is an opportunity to exclude any records that should not contribute to any downstream
-parameter derivations. In the below example this includes only selecting valid assessments and those occurring on or
-after randomization date.
+When deriving `ANL01FL` this is an opportunity to exclude any records that
+should not contribute to any downstream parameter derivations. In the below
+example this includes only selecting valid assessments and those occurring on or
+after randomization date. If there is more than one assessment at a date, the
+worst one is flagged.
 
 ```{r}
 adrs <- adrs %>%
-  mutate(
-    ANL01FL = case_when(
-      !is.na(AVAL) & ADT >= RANDDT ~ "Y",
-      TRUE ~ NA_character_
-    )
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(STUDYID, USUBJID, ADT),
+      order = vars(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "last"
+    ),
+    filter = !is.na(AVAL) & ADT >= RANDDT
```

Comment:
@bundfussr do you think we should also replace this in the ADRS template? i'd say so as i like your code as being more robust a starting point to work from. besides that, all the updates look fine to me

Reply:
Yes, good idea.